optaplanner-docs/src/main/asciidoc/Chapter-Partitioned_search/Chapter-Partitioned_search.adoc

```diff
@@ -8,27 +8,214 @@
+It only works on use cases for which the planning entities and value ranges can be split into n partition,
+such that none of the constraints cross the boundaries between partitions.
 ====
 
 
+[[partitionedSearchConfiguration]]
+== Configuration
+
+Simplest configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+  </partitionedSearch>
+----
+
+There are several ways to <<partitioningASolution,partition a solution>>.
+
+Advanced configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    ...
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+    <threadFactoryClass>...MyAppServerThreadFactory</threadFactoryClass>
+    <runnablePartThreadLimit>4</runnablePartThreadLimit>
+
+    <constructionHeuristic>...</constructionHeuristic>
+    <localSearch>...</localSearch>
+  </partitionedSearch>
+----
+
+The `threadFactoryClass` allows to plug in a custom `ThreadFactory` for environments
+that don't like arbitrary thread creation,
+such as most application servers, Android or Google App Engine.
+
+The `runnablePartThreadLimit` allows limiting CPU usage, see below.
+
+Just like a `<solver>` element, the `<partitionedSearch>` element can one or more <<solverPhase,phases>>.
+Each of those phases will be run on each partition.
+
+A popular configuration is to first run a Partitioned Search phase
+(which includes a Construction Heuristic and a Local Search)
+and then do a non-partitioned Local Search phase:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+
+    <constructionHeuristic/>
+    <localSearch>
+      <secondsSpentLimit>60</secondsSpentLimit>
+    </localSearch>
+  </partitionedSearch>
+  <localSearch/>
+----
+
+
+[[partitioningASolution]]
+== Partitioning a Solution
+
+
+[[customSolutionPartitioner]]
+=== Custom SolutionPartitioner
+
+To use a custom partitioner, configure a partitioner on the Partitioned Search phase:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+  </partitionedSearch>
+----
+
+Implement the `SolutionPartitioner` interface:
+
+[source,java,options="nowrap"]
+----
+public interface SolutionPartitioner<Solution_> {
+
+    List<Solution_> splitWorkingSolution(ScoreDirector<Solution_> scoreDirector);
+
+}
+----
+
+The `size()` of the returned `List` is the `partCount`, so the number of partitions.
```

Comment:
The `size()`` of the returned `List` is the `partCount` (the number of partitions).